File: Python/String/SimplifyPath.py
def SimplifyPath(path):
    result = "/"
    stk = []
    listpath = path.split("/")
    for char in listpath:
        if char == ".." :
            if len(stk) > 0:
                del stk[-1]
        elif char == "." or char =='':
            continue
        
        else:
            stk.append(char)
    
    while len(stk) > 0:
        lastchar = stk[0]
        del stk[0]
        if len(stk) == 0:
            result += lastchar
        else:
            result += lastchar + "/"
    
    return result
  
path = "/../"
result = SimplifyPath(path)
print("result:",result)

path = "/a/./b/../../c/"
result = SimplifyPath(path)
print("result-1:",result)

path = "/home//food"
result = SimplifyPath(path)
print("Result-2:",result)

File: C++/string/Simplify Path.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<string>
#include<sstream>
#include<deque>
#include<memory>

using namespace std;

class Solution
{
    private:
        vector<string>SplitString(string Input);
    public:
        string SimplifyPath(string path)
        {
            vector<string>SplitResult = SplitString(path);
            deque<string>Stack ;
            for (auto x:SplitResult)
            {
                if ( x.empty() || x == ".") 
                    continue;
                else if( x== "..")
                {
                    if (!Stack.empty())
                        {
                           Stack.pop_back();     
                        } 
                }
                else
                {
                    Stack.push_back(x);
                }
            }
            for(auto str: Stack)
            {
                cout<<str<<endl;
            }
            string Result = "/";
            while ( Stack.size() > 0) 
            {
                string Last = Stack.front();
                Stack.pop_front();
                if ( Stack.size() == 0)
                {
                    Result += Last ;
                }
                else
                {
                   Result += Last + "/";  
                   cout<<"Result:"<<Result<<endl;   
                }
            }

            return Result;
        }
};
vector<string>Solution :: SplitString(string Input)
{
    vector<string>Result;
    stringstream ss(Input);
    string Token = "";
    char delimiter = '/';
    while(getline(ss,Token,delimiter))
    {
        Result.emplace_back(Token);
    }
    return Result;
}

int main()
{
    unique_ptr<Solution>s(new Solution);
    string Input = "/home/food";
    auto Result = s->SimplifyPath(Input);
    cout<<Result<<endl;
    Input = "/../";
    Result = s->SimplifyPath(Input);
    cout<<Result<<endl;

    return 0;
}
